internal/provider/provider.go

```diff
@@ -207,25 +246,51 @@ func (p *aapProviderModel) ReadValues(host, username, password *string, insecure
 	if *host == "" {
 		*host = os.Getenv("AAP_HOST")
 	}
-	*username = os.Getenv("AAP_USERNAME")
-	*password = os.Getenv("AAP_PASSWORD")
-
-	*insecureSkipVerify = DefaultInsecureSkipVerify
-	var err error
 
 	// Read host from user configuration
 	if !p.Host.IsNull() {
 		*host = p.Host.ValueString()
 	}
-	// Read username from user configuration
-	if !p.Username.IsNull() {
-		*username = p.Username.ValueString()
+
+	*token = os.Getenv("AAP_TOKEN")
+	if !p.Token.IsNull() {
+		// Read token from user configuration
+		*token = p.Token.ValueString()
 	}
-	// Read password from user configuration
-	if !p.Password.IsNull() {
-		*password = p.Password.ValueString()
+
+	if len(*token) > 0 {
+		if !p.Username.IsNull() {
+			resp.Diagnostics.AddAttributeWarning(
+				path.Root("username"),
+				"Inconsistent configuration for username",
+				"When token is configured for authentication, username will be ignored. Please remove username from your configuration.",
+			)
+		}
+		if !p.Password.IsNull() {
+			resp.Diagnostics.AddAttributeWarning(
+				path.Root("username"),
```

Comment:
path.Root("username") should be path.Root("password")

Reply:
bc8b259b0ef592e75f27a43257c80d60a04bc16d

Reply:
Great catch!